src/libsystemd/sd-bus/bus-message.c

```diff
@@ -4667,3 +4667,26 @@ _public_ int sd_bus_message_sensitive(sd_bus_message *m) {
         m->sensitive = true;
         return 0;
 }
+
+char** bus_message_make_log_fields(sd_bus_message *m) {
+        _cleanup_strv_free_ char **strv = NULL;
+
+        assert(m);
+
+        (void) strv_extend_join(&strv, "DBUS_MESSAGE_TYPE=", bus_message_type_to_string(m->header->type));
+        (void) strv_extend_join(&strv, "DBUS_SENDER=", sd_bus_message_get_sender(m));
+        (void) strv_extend_join(&strv, "DBUS_DESTINATION=", sd_bus_message_get_destination(m));
+        (void) strv_extend_join(&strv, "DBUS_PATH=", sd_bus_message_get_path(m));
+        (void) strv_extend_join(&strv, "DBUS_INTERFACE=", sd_bus_message_get_interface(m));
+        (void) strv_extend_join(&strv, "DBUS_MEMBER=", sd_bus_message_get_member(m));
+
+        (void) strv_extendf(&strv, "DBUS_MESSAGE_COOKIE=%" PRIu64, BUS_MESSAGE_COOKIE(m));
+        if (m->reply_cookie != 0)
+                (void) strv_extendf(&strv, "DBUS_MESSAGE_REPLY_COOKIE=%" PRIu64, m->reply_cookie);
+
+        (void) strv_extend_join(&strv, "DBUS_SIGNATURE=", m->root_container.signature);
+        (void) strv_extend_join(&strv, "DBUS_ERROR_NAME=", m->error.name);
+        (void) strv_extend_join(&strv, "DBUS_ERROR_MESSAGE=", m->error.message);
+
+        return TAKE_PTR(strv);
+}
```

Comment:
instead of enabling this manually in all our code, shouldn't this be something we just enable when `DEBUG_LOGGING` holds?

That way, once you enable debug logging you get the extra fields for *all* messages just like that, but without it we won't waste time on all those memory allocations?

Reply:
or here's another idea, instead of adding an boolean settng to the sd_bus object, and another one to sd_device and so on – maybe instead add one to log.c called "log_request_context_requested" or so that sd-bus, sd-device and so on all check on their own. i.e. a single global option I think is much nicer than having to enable this in every subsystem individually.

Reply:
or maybe make log_request_context_requested a tristate:

1. if 0: no extended log contexts are added anywhere
2. if > 0: extended log contexts are added everywhere
3. if < 0: extended log contexts are done if debug logging is on

< 0 would be the default of course.

And then maybe add an env var for this too, so that callers can request this from the command lines explicitly without necessarily having to enable debug logging.